### ISSUE TITLE
[Basisu] MinGW build failure

### ISSUE THREAD

**Issue by addowhite** [microsoft/vcpkg#27759]
# TLDR
Basisu won't build with mingw triplet due to not including `libgen.h` on Windows.
However, the libgen include is required on Windows when using MinGW as it's needed for definitions of `dirname()` and `basename()`.

The culprit:
```c++
// src/encoder/basisu_enc.h
#ifndef _WIN32
#include <libgen.h>
#endif
```

Recommend changing to:
```c++
// src/encoder/basisu_enc.h
#if !defined(_WIN32) or defined(__MINGW32__)
#include <libgen.h>
#endif
```

---

Package: basisu[core]:x64-mingw-dynamic -> 1.11#6

**Host Environment**

- Host: x64-windows
- Compiler: GNU 10.3.0
-    vcpkg-tool version: 2022-10-17-3247920fbdd47d08f36cbd480addd9890d3c2435
    vcpkg-scripts version: 68ad399d5 2022-11-04 (6 days ago)

**To Reproduce**

`vcpkg install basisu --triplet=x64-mingw-dynamic`

**Failure logs**

```
-- Using cached jherico-basis_universal-497875f756ed0e3eb62e0ff08d55c62242f4be74.tar.gz.
-- Cleaning sources at C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/vcpkg/downloads/jherico-basis_universal-497875f756ed0e3eb62e0ff08d55c62242f4be74.tar.gz
-- Applying patch fix-addostream.patch
-- Using source at C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean
-- Configuring x64-mingw-dynamic
-- Building x64-mingw-dynamic-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:131 (message):
    Command failed: C:/vcpkg/downloads/tools/cmake-3.24.0-windows/cmake-3.24.0-windows-i386/bin/cmake.exe --build . --config Debug --target install -- -v -j9
    Working Directory: C:/vcpkg/buildtrees/basisu/x64-mingw-dynamic-dbg
    See logs for more information:
      C:\vcpkg\buildtrees\basisu\install-x64-mingw-dynamic-dbg-out.log

Call Stack (most recent call first):
  installed/x64-mingw-dynamic/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  installed/x64-mingw-dynamic/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/basisu/portfile.cmake:17 (vcpkg_cmake_install)
  scripts/ports.cmake:147 (include)



```
<details><summary>C:\vcpkg\buildtrees\basisu\install-x64-mingw-dynamic-dbg-out.log</summary>

```
[1/19] C:\msys64\mingw64\bin\x86_64-w64-mingw32-g++.exe -DBASISU_NO_ITERATOR_DEBUG_LEVEL -Dbasisu_encoder_EXPORTS -IC:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/.. -IC:/vcpkg/installed/x64-mingw-dynamic/include -IC:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/transcoder -g -MD -MT src/encoder/CMakeFiles/basisu_encoder.dir/basisu_astc_decomp.cpp.obj -MF src\encoder\CMakeFiles\basisu_encoder.dir\basisu_astc_decomp.cpp.obj.d -o src/encoder/CMakeFiles/basisu_encoder.dir/basisu_astc_decomp.cpp.obj -c C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_astc_decomp.cpp
[2/19] C:\msys64\mingw64\bin\x86_64-w64-mingw32-g++.exe -DBASISU_NO_ITERATOR_DEBUG_LEVEL -Dbasisu_encoder_EXPORTS -IC:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/.. -IC:/vcpkg/installed/x64-mingw-dynamic/include -IC:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/transcoder -g -MD -MT src/encoder/CMakeFiles/basisu_encoder.dir/basisu_etc.cpp.obj -MF src\encoder\CMakeFiles\basisu_encoder.dir\basisu_etc.cpp.obj.d -o src/encoder/CMakeFiles/basisu_encoder.dir/basisu_etc.cpp.obj -c C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_etc.cpp
FAILED: src/encoder/CMakeFiles/basisu_encoder.dir/basisu_etc.cpp.obj 
C:\msys64\mingw64\bin\x86_64-w64-mingw32-g++.exe -DBASISU_NO_ITERATOR_DEBUG_LEVEL -Dbasisu_encoder_EXPORTS -IC:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/.. -IC:/vcpkg/installed/x64-mingw-dynamic/include -IC:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/transcoder -g -MD -MT src/encoder/CMakeFiles/basisu_encoder.dir/basisu_etc.cpp.obj -MF src\encoder\CMakeFiles\basisu_encoder.dir\basisu_etc.cpp.obj.d -o src/encoder/CMakeFiles/basisu_encoder.dir/basisu_etc.cpp.obj -c C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_etc.cpp
In file included from C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_etc.h:16,
                 from C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_etc.cpp:15:
C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_enc.h: In function 'bool basisu::string_split_path(const char*, std::string*, std::string*, std::string*, std::string*)':
C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_enc.h:749:26: error: 'dirname' was not declared in this scope
  749 |   const char *pDirName = dirname(dirtmp);
      |                          ^~~~~~~
C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_enc.h:750:27: error: 'basename' was not declared in this scope; did you mean 'pBaseName'?
  750 |   const char* pBaseName = basename(nametmp);
      |                           ^~~~~~~~
      |                           pBaseName
[3/19] C:\msys64\mingw64\bin\x86_64-w64-mingw32-g++.exe -DBASISU_NO_ITERATOR_DEBUG_LEVEL -Dbasisu_encoder_EXPORTS -IC:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/.. -IC:/vcpkg/installed/x64-mingw-dynamic/include -IC:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/transcoder -g -MD -MT src/encoder/CMakeFiles/basisu_encoder.dir/basisu_global_selector_palette_helpers.cpp.obj -MF src\encoder\CMakeFiles\basisu_encoder.dir\basisu_global_selector_palette_helpers.cpp.obj.d -o src/encoder/CMakeFiles/basisu_encoder.dir/basisu_global_selector_palette_helpers.cpp.obj -c C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_global_selector_palette_helpers.cpp
...
Skipped 18 lines
...

C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_enc.h: In function 'bool basisu::string_split_path(const char*, std::string*, std::string*, std::string*, std::string*)':
C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_enc.h:749:26: error: 'dirname' was not declared in this scope
  749 |   const char *pDirName = dirname(dirtmp);
      |                          ^~~~~~~
C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_enc.h:750:27: error: 'basename' was not declared in this scope; did you mean 'pBaseName'?
  750 |   const char* pBaseName = basename(nametmp);
      |                           ^~~~~~~~
      |                           pBaseName
[5/19] C:\msys64\mingw64\bin\x86_64-w64-mingw32-g++.exe -DBASISU_NO_ITERATOR_DEBUG_LEVEL -Dbasisu_encoder_EXPORTS -IC:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/.. -IC:/vcpkg/installed/x64-mingw-dynamic/include -IC:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/transcoder -g -MD -MT src/encoder/CMakeFiles/basisu_encoder.dir/basisu_basis_file.cpp.obj -MF src\encoder\CMakeFiles\basisu_encoder.dir\basisu_basis_file.cpp.obj.d -o src/encoder/CMakeFiles/basisu_encoder.dir/basisu_basis_file.cpp.obj -c C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_basis_file.cpp
FAILED: src/encoder/CMakeFiles/basisu_encoder.dir/basisu_basis_file.cpp.obj 
C:\msys64\mingw64\bin\x86_64-w64-mingw32-g++.exe -DBASISU_NO_ITERATOR_DEBUG_LEVEL -Dbasisu_encoder_EXPORTS -IC:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/.. -IC:/vcpkg/installed/x64-mingw-dynamic/include -IC:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/transcoder -g -MD -MT src/encoder/CMakeFiles/basisu_encoder.dir/basisu_basis_file.cpp.obj -MF src\encoder\CMakeFiles\basisu_encoder.dir\basisu_basis_file.cpp.obj.d -o src/encoder/CMakeFiles/basisu_encoder.dir/basisu_basis_file.cpp.obj -c C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_basis_file.cpp
In file included from C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_backend.h:17,
                 from C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_basis_file.h:17,
                 from C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_basis_file.cpp:15:
C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_enc.h: In function 'bool basisu::string_split_path(const char*, std::string*, std::string*, std::string*, std::string*)':
C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_enc.h:749:26: error: 'dirname' was not declared in this scope
  749 |   const char *pDirName = dirname(dirtmp);
      |                          ^~~~~~~
C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_enc.h:750:27: error: 'basename' was not declared in this scope; did you mean 'pBaseName'?
  750 |   const char* pBaseName = basename(nametmp);
      |                           ^~~~~~~~
      |                           pBaseName
[6/19] C:\msys64\mingw64\bin\x86_64-w64-mingw32-g++.exe -DBASISU_NO_ITERATOR_DEBUG_LEVEL -Dbasisu_encoder_EXPORTS -IC:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/.. -IC:/vcpkg/installed/x64-mingw-dynamic/include -IC:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/transcoder -g -MD -MT src/encoder/CMakeFiles/basisu_encoder.dir/basisu_comp.cpp.obj -MF src\encoder\CMakeFiles\basisu_encoder.dir\basisu_comp.cpp.obj.d -o src/encoder/CMakeFiles/basisu_encoder.dir/basisu_comp.cpp.obj -c C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_comp.cpp
FAILED: src/encoder/CMakeFiles/basisu_encoder.dir/basisu_comp.cpp.obj 
C:\msys64\mingw64\bin\x86_64-w64-mingw32-g++.exe -DBASISU_NO_ITERATOR_DEBUG_LEVEL -Dbasisu_encoder_EXPORTS -IC:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/.. -IC:/vcpkg/installed/x64-mingw-dynamic/include -IC:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/transcoder -g -MD -MT src/encoder/CMakeFiles/basisu_encoder.dir/basisu_comp.cpp.obj -MF src\encoder\CMakeFiles\basisu_encoder.dir\basisu_comp.cpp.obj.d -o src/encoder/CMakeFiles/basisu_encoder.dir/basisu_comp.cpp.obj -c C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_comp.cpp
In file included from C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_frontend.h:16,
                 from C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_comp.h:16,
                 from C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_comp.cpp:15:
C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_enc.h: In function 'bool basisu::string_split_path(const char*, std::string*, std::string*, std::string*, std::string*)':
C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_enc.h:749:26: error: 'dirname' was not declared in this scope
  749 |   const char *pDirName = dirname(dirtmp);
      |                          ^~~~~~~
C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_enc.h:750:27: error: 'basename' was not declared in this scope; did you mean 'pBaseName'?
  750 |   const char* pBaseName = basename(nametmp);
      |                           ^~~~~~~~
      |                           pBaseName
[7/19] C:\msys64\mingw64\bin\x86_64-w64-mingw32-g++.exe -DBASISU_NO_ITERATOR_DEBUG_LEVEL -Dbasisu_encoder_EXPORTS -IC:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/.. -IC:/vcpkg/installed/x64-mingw-dynamic/include -IC:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/transcoder -g -MD -MT src/encoder/CMakeFiles/basisu_encoder.dir/basisu_enc.cpp.obj -MF src\encoder\CMakeFiles\basisu_encoder.dir\basisu_enc.cpp.obj.d -o src/encoder/CMakeFiles/basisu_encoder.dir/basisu_enc.cpp.obj -c C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_enc.cpp
FAILED: src/encoder/CMakeFiles/basisu_encoder.dir/basisu_enc.cpp.obj 
C:\msys64\mingw64\bin\x86_64-w64-mingw32-g++.exe -DBASISU_NO_ITERATOR_DEBUG_LEVEL -Dbasisu_encoder_EXPORTS -IC:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/.. -IC:/vcpkg/installed/x64-mingw-dynamic/include -IC:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/transcoder -g -MD -MT src/encoder/CMakeFiles/basisu_encoder.dir/basisu_enc.cpp.obj -MF src\encoder\CMakeFiles\basisu_encoder.dir\basisu_enc.cpp.obj.d -o src/encoder/CMakeFiles/basisu_encoder.dir/basisu_enc.cpp.obj -c C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_enc.cpp
In file included from C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_enc.cpp:15:
C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_enc.h: In function 'bool basisu::string_split_path(const char*, std::string*, std::string*, std::string*, std::string*)':
C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_enc.h:749:26: error: 'dirname' was not declared in this scope
  749 |   const char *pDirName = dirname(dirtmp);
      |                          ^~~~~~~
C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_enc.h:750:27: error: 'basename' was not declared in this scope; did you mean 'pBaseName'?
  750 |   const char* pBaseName = basename(nametmp);
      |                           ^~~~~~~~
      |                           pBaseName
[8/19] C:\msys64\mingw64\bin\x86_64-w64-mingw32-g++.exe -DBASISU_NO_ITERATOR_DEBUG_LEVEL -Dbasisu_encoder_EXPORTS -IC:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/.. -IC:/vcpkg/installed/x64-mingw-dynamic/include -IC:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/transcoder -g -MD -MT src/encoder/CMakeFiles/basisu_encoder.dir/basisu_frontend.cpp.obj -MF src\encoder\CMakeFiles\basisu_encoder.dir\basisu_frontend.cpp.obj.d -o src/encoder/CMakeFiles/basisu_encoder.dir/basisu_frontend.cpp.obj -c C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_frontend.cpp
FAILED: src/encoder/CMakeFiles/basisu_encoder.dir/basisu_frontend.cpp.obj 
C:\msys64\mingw64\bin\x86_64-w64-mingw32-g++.exe -DBASISU_NO_ITERATOR_DEBUG_LEVEL -Dbasisu_encoder_EXPORTS -IC:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/.. -IC:/vcpkg/installed/x64-mingw-dynamic/include -IC:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/transcoder -g -MD -MT src/encoder/CMakeFiles/basisu_encoder.dir/basisu_frontend.cpp.obj -MF src\encoder\CMakeFiles\basisu_encoder.dir\basisu_frontend.cpp.obj.d -o src/encoder/CMakeFiles/basisu_encoder.dir/basisu_frontend.cpp.obj -c C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_frontend.cpp
In file included from C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_frontend.h:16,
                 from C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_frontend.cpp:20:
C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_enc.h: In function 'bool basisu::string_split_path(const char*, std::string*, std::string*, std::string*, std::string*)':
C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_enc.h:749:26: error: 'dirname' was not declared in this scope
  749 |   const char *pDirName = dirname(dirtmp);
      |                          ^~~~~~~
C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_enc.h:750:27: error: 'basename' was not declared in this scope; did you mean 'pBaseName'?
  750 |   const char* pBaseName = basename(nametmp);
      |                           ^~~~~~~~
      |                           pBaseName
[9/19] C:\msys64\mingw64\bin\x86_64-w64-mingw32-g++.exe -DBASISU_NO_ITERATOR_DEBUG_LEVEL -Dbasisu_encoder_EXPORTS -IC:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/.. -IC:/vcpkg/installed/x64-mingw-dynamic/include -IC:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/transcoder -g -MD -MT src/encoder/CMakeFiles/basisu_encoder.dir/basisu_pvrtc1_4.cpp.obj -MF src\encoder\CMakeFiles\basisu_encoder.dir\basisu_pvrtc1_4.cpp.obj.d -o src/encoder/CMakeFiles/basisu_encoder.dir/basisu_pvrtc1_4.cpp.obj -c C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_pvrtc1_4.cpp
FAILED: src/encoder/CMakeFiles/basisu_encoder.dir/basisu_pvrtc1_4.cpp.obj 
C:\msys64\mingw64\bin\x86_64-w64-mingw32-g++.exe -DBASISU_NO_ITERATOR_DEBUG_LEVEL -Dbasisu_encoder_EXPORTS -IC:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/.. -IC:/vcpkg/installed/x64-mingw-dynamic/include -IC:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/transcoder -g -MD -MT src/encoder/CMakeFiles/basisu_encoder.dir/basisu_pvrtc1_4.cpp.obj -MF src\encoder\CMakeFiles\basisu_encoder.dir\basisu_pvrtc1_4.cpp.obj.d -o src/encoder/CMakeFiles/basisu_encoder.dir/basisu_pvrtc1_4.cpp.obj -c C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_pvrtc1_4.cpp
In file included from C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_etc.h:16,
                 from C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_gpu_texture.h:16,
                 from C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_pvrtc1_4.h:16,
                 from C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_pvrtc1_4.cpp:15:
C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_enc.h: In function 'bool basisu::string_split_path(const char*, std::string*, std::string*, std::string*, std::string*)':
C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_enc.h:749:26: error: 'dirname' was not declared in this scope
  749 |   const char *pDirName = dirname(dirtmp);
      |                          ^~~~~~~
C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/encoder/basisu_enc.h:750:27: error: 'basename' was not declared in this scope; did you mean 'pBaseName'?
  750 |   const char* pBaseName = basename(nametmp);
      |                           ^~~~~~~~
      |                           pBaseName
[10/19] C:\msys64\mingw64\bin\x86_64-w64-mingw32-g++.exe -DBASISU_NO_ITERATOR_DEBUG_LEVEL -Dbasisu_transcoder_EXPORTS -IC:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/transcoder -g -MD -MT src/transcoder/CMakeFiles/basisu_transcoder.dir/basisu_transcoder.cpp.obj -MF src\transcoder\CMakeFiles\basisu_transcoder.dir\basisu_transcoder.cpp.obj.d -o src/transcoder/CMakeFiles/basisu_transcoder.dir/basisu_transcoder.cpp.obj -c C:/vcpkg/buildtrees/basisu/src/2242f4be74-920b5d9314.clean/src/transcoder/basisu_transcoder.cpp
ninja: build stopped: subcommand failed.
```
</details>

**Additional context**
